app/uk/gov/hmrc/fileupload/controllers/AdminController.scala

```diff
@@ -16,16 +16,41 @@
 
 package uk.gov.hmrc.fileupload.controllers
 
+import play.api.Logger
+import play.api.libs.json.{JsString, Json}
 import play.api.mvc.{Action, Controller}
 import uk.gov.hmrc.fileupload.notifier.NotifierService._
+import uk.gov.hmrc.fileupload.quarantine.FileInfo
 import uk.gov.hmrc.fileupload.transfer.TransferRequested
+import uk.gov.hmrc.fileupload.utils.errorAsJson
 import uk.gov.hmrc.fileupload.virusscan.VirusScanRequested
 import uk.gov.hmrc.fileupload.{EnvelopeId, FileId, FileRefId}
 
 import scala.concurrent.{ExecutionContext, Future}
+import scala.util.control.NonFatal
 
-class AdminController (notify: AnyRef => Future[NotifyResult])
-                      (implicit executionContext: ExecutionContext) extends Controller {
+class AdminController(getFileInfo: (FileRefId) => Future[Option[FileInfo]], getChunks: (FileRefId) => Future[Int])(notify: AnyRef => Future[NotifyResult])
+                     (implicit executionContext: ExecutionContext) extends Controller {
+
+  def fileInfo(fileRefId: FileRefId) = Action.async { request =>
+    getFileInfo(fileRefId).flatMap {
+      case Some(f) => {
+        val expectedNoChunks = math.ceil(f.length.toDouble / f.chunkSize)
+        getChunks(fileRefId).map { actualNoChunks =>
+          if (actualNoChunks == expectedNoChunks) {
+            Ok(Json.toJson(f))
+          } else {
+            Ok(errorAsJson(s"Some file chunks are missing! Number of chunks expected $expectedNoChunks , actual $actualNoChunks"))
+          }
+        }.recover {
+          case NonFatal(ex) =>
+            Logger.warn(s"Retrieval of chunks for the file id $fileRefId failed ${ex.getMessage}")
+            InternalServerError(ex.getMessage)
+        }
+      }
+      case None => Future.successful(NotFound)
```

Comment:
we could still check how many chunks there are? we could also do that as another ticket... what do you think?